File: typecast-js/src/types/Subscription.ts
/**
 * Subscription plan tier returned by `GET /v1/users/me/subscription`.
 */
export type PlanTier = 'free' | 'lite' | 'plus' | 'custom';

/**
 * Credit usage information.
 */
export interface Credits {
  /** Total credits provided by the plan. */
  plan_credits: number;
  /** Number of credits used. */
  used_credits: number;
}

/**
 * Usage limit information.
 */
export interface Limits {
  /** Maximum number of concurrent requests allowed. */
  concurrency_limit: number;
}

/**
 * Response from `GET /v1/users/me/subscription`.
 */
export interface SubscriptionResponse {
  /** Current subscription plan tier. */
  plan: PlanTier;
  /** Credit usage information. */
  credits: Credits;
  /** Usage limit information. */
  limits: Limits;
}

File: typecast-python/src/typecast/models/subscription.py
from enum import Enum

from pydantic import BaseModel, Field


class PlanTier(str, Enum):
    """Subscription plan tier."""

    FREE = "free"
    LITE = "lite"
    PLUS = "plus"
    CUSTOM = "custom"


class Credits(BaseModel):
    """Credit usage information."""

    plan_credits: int = Field(description="Total credits provided by the plan")
    used_credits: int = Field(description="Number of credits used")


class Limits(BaseModel):
    """Usage limit information."""

    concurrency_limit: int = Field(
        description="Maximum number of concurrent requests allowed"
    )


class SubscriptionResponse(BaseModel):
    """Response from `GET /v1/users/me/subscription`."""

    plan: PlanTier = Field(description="Current subscription plan tier")
    credits: Credits = Field(description="Credit usage information")
    limits: Limits = Field(description="Usage limit information")
